### PR TITLE
Fix support for MessageHeaderArrayAttribute and MessagePropertyAttribute

### LINKED ISSUE #558: `attributeType` checked twice against `MessageBodyMemberAttribute`
`attributeType` is checked against `MessageBodyMemberAttribute` on both line 60 and 68.
https://github.com/CoreWCF/CoreWCF/blob/73ac30239b8f8eaea5af23d09c8e8d4405ee86b7/src/CoreWCF.Primitives/src/CoreWCF/Description/CustomAttributeProvider.cs#L60-L75

As the second branch uses `ServiceReflector.SMMessagePropertyAttributeFullName` I guess the second statement should have used `MessagePropertyAttribute` instead.

### PR DESCRIPTION
**Pull request by mconnew**
Fixes #558 and a problem I found with MessageHeaderArrayAttribute at the same time.

### CODE REVIEW

File: src/CoreWCF.Primitives/src/CoreWCF/Description/MessagePropertyDescriptionCollection.cs [hunk 1]
```diff
@@ -6,6 +6,6 @@ namespace CoreWCF.Description
     public class MessagePropertyDescriptionCollection : System.Collections.ObjectModel.KeyedCollection<string, MessagePropertyDescription>
     {
         internal MessagePropertyDescriptionCollection() { }
-        protected override string GetKeyForItem(MessagePropertyDescription item) { return default; }
+        protected override string GetKeyForItem(MessagePropertyDescription item) => item.Name;
```
Comment (jnyrup) [inline src/CoreWCF.Primitives/src/CoreWCF/Description/MessagePropertyDescriptionCollection.cs:9]: I don't know if relevant, but these other collections use `return default`
* `MessagePartDescriptionCollection`
* `MessageHeaderDescriptionCollection`
* `MessageDescriptionCollection`

File: src/CoreWCF.Primitives/src/CoreWCF/Description/MessageContractHelper.cs [hunk 2]
```diff
@@ -23,21 +23,23 @@ internal static bool IsMessageContract(Type type)
...
-                , ServiceReflector.CWCFMesssageHeaderArrayAttribute
-                ,ServiceReflector.CWCFMesssageBodyMemberAttribute
-                ,ServiceReflector.CWCFMesssagePropertyAttribute
-                ,ServiceReflector.SMMessageBodyMemberAttributeFullName
-                , ServiceReflector.SMMessageContractAttributeFullName
-                ,ServiceReflector.SMMessageHeaderAttributeFullName
+                ServiceReflector.CWCFMesssageHeaderAttribute,
+                ServiceReflector.CWCFMesssageHeaderArrayAttribute,
+                ServiceReflector.CWCFMesssageBodyMemberAttribute,
+                ServiceReflector.CWCFMesssagePropertyAttribute,
```
[the remark this answers is not in the document]
Comment (mconnew) [reply src/CoreWCF.Primitives/src/CoreWCF/Description/MessageContractHelper.cs:31]: Oops, I'll fix that. Thanks.

File: src/CoreWCF.Primitives/src/CoreWCF/Description/MessagePropertyDescriptionCollection.cs (hunk 1 above)
Comment (mconnew) [reply src/CoreWCF.Primitives/src/CoreWCF/Description/MessagePropertyDescriptionCollection.cs:9]: That does sound like a problem. I'll check against .NET Framework and make them match.

**Comment by mconnew** [comment]
@jnyrup, I checked every collection type we have and made sure they were all correct. I found another non-header related one which needed fixing too.

**Pull request merged**